src/data/mystery-encounters/encounters/choice-of-balance-encounter.ts

```diff
@@ -0,0 +1,393 @@
+    // Leave encounter with no rewards or exp
+      scene.mysteryEncounterAuras.AddAura(scene.currentBattle.mysteryEncounter.misc[1].negativeAura);
+      scene.mysteryEncounterAuras.AddAura(scene.currentBattle.mysteryEncounter.misc[1].positiveAura);
+      leaveEncounterWithoutBattle(scene, true);
+      return true;
+    })
+  .withSimpleOption(
+    {
+      buttonLabel: `${namespace}_option_3_label`,
+      buttonTooltip: `${namespace}_dynamic_option_3`,
+      selected: [
+        {
+          text: `${namespace}_option_selected_message`,
+        },
+      ],
+    },
+    async (scene: BattleScene) => {
+      // Leave encounter with no rewards or exp
+      scene.mysteryEncounterAuras.AddAura(scene.currentBattle.mysteryEncounter.misc[2].negativeAura);
+      scene.mysteryEncounterAuras.AddAura(scene.currentBattle.mysteryEncounter.misc[2].positiveAura);
+      leaveEncounterWithoutBattle(scene, true);
+      return true;
+    })
+  .withSimpleOption(
+    {
+      buttonLabel: `${namespace}_option_4_label`,
+      buttonTooltip: `${namespace}_option_4_description`,
+      selected: [
+        {
+          text: `${namespace}_option_leave_selected_message`,
+        },
+      ],
+    },
+    async (scene: BattleScene) => {
+      // Leave encounter with no rewards or exp
+      leaveEncounterWithoutBattle(scene, true);
+      return true;
+    })
+  .build();
+
+export enum NegativeRewards {
+  INCOME,
+  LUCK,
+  PLAYER_STATS,
+  ENEMY_STATS,
+  PP,
+  NO_REROLL
+}
+
+export enum PositiveRewards {
+  INCOME = 1000, // starting at 1000 for positive so we can tell them apart between the negatives later
+  LUCK,
+  PLAYER_STATS,
+  ENEMY_STATS,
+  PP,
+  INSTANT_MONEY,
+  INSTANT_CANDY
+}
+
+export function getNegativeRewards(totalOptions: number): number[] {
+  const negativeOptions = [];
+  const numNegatives = Object.keys(NegativeRewards).filter(nr => !isNaN(Number(nr)));
+  while (negativeOptions.length < totalOptions) {
+    const roll = randSeedInt(numNegatives.length);
+    if (!negativeOptions.includes(roll)) {
+      negativeOptions.push(roll);
+    }
+  }
+  return negativeOptions;
+}
+
+export function getPositiveRewards(totalOptions: number): number[] {
+  const positiveOptions = [];
+  const numPositives = Object.values(PositiveRewards).filter(pr => !isNaN(Number(pr)));
+  while (positiveOptions.length < totalOptions) {
+    const roll = randSeedInt(numPositives.length);
+    if (!positiveOptions.includes(numPositives[roll])) {
+      positiveOptions.push(numPositives[roll]);
+    }
+  }
+  return positiveOptions;
+}
+
+export class RewardOption {
+  public negativeOption: number;
+  public positiveOption: number;
+  public negativeStat = -1;
+  public positiveStat = -1;
+  public negativeAura: Aura;
+  public positiveAura: Aura;
+
+  constructor(negativeOption: number, positiveOption: number) {
+    this.negativeOption = negativeOption;
+    this.positiveOption = positiveOption;
+  }
+
+  generateMessage(): string {
```

Comment:
This type of function is going to probably cause lots of issues with translations. Could we instead just have a few properly formatted string templates that are stored in the locales file? And you can inject the programmatic logic in via token instead

Reply:
Examples of a string template:
```
"augmentTextTemplate": "{{descriptor}} for {{duration}}"
"augmentTextJoiner": ", then "
```

etc.

Reply:
and to get the proper string, you would just do `i18next.t("mysteryEncounter:augmentTextTemplate", { descriptor: this.getDescriptorText(negativeText), duration: this.formattedWaves(negativeDuration) });`